test/e2e/accordion.cy.js

```diff
@@ -0,0 +1,40 @@
+describe('Accordion', function () {
+  function loopThroughAccordion(accordionComponent) {
+    const itemsCount = Object.keys(accordionComponent._items).length
+    cy.get('.accordion-item').should('have.length', itemsCount)
+    for (let i = 0; i < itemsCount; i++) {
+      const bodyWithoutHtml = accordionComponent._items[i].body.replace(/<[^>]*>/g, '')
+      cy.get('.accordion-item').eq(i).within(() => {
+        cy.get('.accordion-item__btn.is-visited').should('not.exist')
+        cy.get('.accordion-item__body-inner').should('not.be.visible')
+        cy.get('.accordion-item__title').should('contain', accordionComponent._items[i].title).click()
+        cy.get('.accordion-item__btn.is-visited').should('exist')
+        cy.get('.accordion-item__body-inner')
+          .should('be.visible')
+          .should('contain', bodyWithoutHtml)
+        cy.get('.accordion-item__title').click()
+        cy.get('.accordion-item__body-inner').should('not.be.visible')
+      })
```

Comment:
Could we use a forEach here instead of index iterating?

```suggestion
    const items = accordionComponent._items
    cy.get('.accordion-item').should('have.length', items.length)
    items.forEach((item) =>  {
      const bodyWithoutHtml = item.body.replace(/<[^>]*>/g, '')
      cy.get('.accordion-item').eq(i).within(() => {
        cy.get('.accordion-item__btn.is-visited').should('not.exist')
        cy.get('.accordion-item__body-inner').should('not.be.visible')
        cy.get('.accordion-item__title').should('contain', item.title).click()
        cy.get('.accordion-item__btn.is-visited').should('exist')
        cy.get('.accordion-item__body-inner')
          .should('be.visible')
          .should('contain', bodyWithoutHtml)
        cy.get('.accordion-item__title').click()
        cy.get('.accordion-item__body-inner').should('not.be.visible')
      })
```

Reply:
We'd still need the index value to make sure we go into the correct item

Reply:
The forEach loop is dictating the single item to examine though? `item.` here is the same as `accordionComponent._items[i]` from before

Reply:
`cy.get('.accordion-item').eq(i).within(() => {`

We'd still need to make sure the tests are looking within the correct accordion list item. This is to make sure the correct is-visited class is applied and as a whole making sure we're checking that item and not another one.

Reply:
Done